File: Java/defend/Defend.java
/*
 * Team 3
 * AJ Downey, Joshua Barbee, Wei Wei Chien
 */

package defend;

import java.util.List;

public class Defend {

    public static void main(String[] args) {
        if (InputManager.ensureNoRedirection()) {
            final String[] name = NameValidator.read();

            final int[] ints = IntsValidator.read();
            final long sum = IntsValidator.add(ints);
            final long product = IntsValidator.multiply(ints);

            final List<String> input = FileIOValidator.readInputFile();

            PasswordValidator.readAndVerify();

            FileIOValidator.readAndPrintToOutputFile(
                    name, ints, sum, product, input
            );
        }
    }
}

File: CS/Defend.cs
/*
 * Team 3
 * AJ Downey, Joshua Barbee, Wei Wei Chien
 */

namespace Defend483CS
{
    internal class Defend
    {

        static void Main(string[] args)
        {
            if (InputManager.EnsureNoRedirection())
            {
                (string firstName, string lastName) = NameValidator.Read();

                (int firstInt, int secondInt, long sum, long product) =
                    IntsValidator.ReadAndAddAndMultiply();

                (string inputFileName, string inputFileContents) =
                    FileIOValidator.ReadInputFile();

                PasswordValidator.ReadAndStoreAndVerify();

                FileIOValidator.ReadFilenameAndPrintToFile(
                    firstName, lastName,
                    firstInt, secondInt, sum, product,
                    inputFileName, inputFileContents
                );
            }
        }
    }
}

